bitcoin/src/blockdata/script/types/borrowed.rs

```diff
@@ -0,0 +1,513 @@
+
+use secp256k1::{Secp256k1, Verification};
+
+use crate::address::WitnessVersion;
+use crate::blockdata::opcodes::{self, all::*};
+use crate::blockdata::script::{bytes_to_asm_fmt, Builder, Instruction, Instructions, InstructionIndices, ScriptBuf};
+#[cfg(feature = "bitcoinconsensus")]
+use crate::blockdata::script::Error;
+use crate::consensus::Encodable;
+use crate::hash_types::{ScriptHash, WScriptHash};
+use crate::hashes::Hash;
+use crate::key::PublicKey;
+use crate::policy::DUST_RELAY_TX_FEE;
+use crate::prelude::*;
+use crate::schnorr::UntweakedPublicKey;
+use crate::taproot::{LeafVersion, TapNodeHash, TapLeafHash};
+
+/// Bitcoin script slice.
+///
+/// *[See also the `bitcoin::blockdata::script` module](crate::blockdata::script).*
+///
+/// `Script` is a script slice, the most primitive script type. It's usually seen in its borrowed
+/// form `&Script`. It is always encoded as a series of bytes representing the opcodes and data
+/// pushes.
+///
+/// ## Validity
+///
+/// `Script` does not have any validity invariants - it's essentially just a marked slice of
+/// bytes. This is similar to [`Path`](std::path::Path) vs [`OsStr`](std::ffi::OsStr) where they
+/// are trivially cast-able to each-other and `Path` doesn't guarantee being a usable FS path but
+/// having a newtype still has value because of added methods, readability and basic type checking.
+///
+/// Although at least data pushes could be checked not to overflow the script, bad scripts are
+/// allowed to be in a transaction (outputs just become unspendable) and there even are such
+/// transactions in the chain. Thus we must allow such scripts to be placed in the transaction.
+///
+/// ## Slicing safety
+///
+/// Slicing is similar to how `str` works: some ranges may be incorrect and indexing by
+/// `usize` is not supported. However, as opposed to `std`, we have no way of checking
+/// correctness without causing linear complexity so there are **no panics on invalid
+/// ranges!** If you supply an invalid range, you'll get a garbled script.
+///
+/// The range is considered valid if it's at a boundary of instruction. Care must be taken
+/// especially with push operations because you could get a reference to arbitrary
+/// attacker-supplied bytes that look like a valid script.
+///
+/// It is recommended to use `.instructions()` method to get an iterator over script
+/// instructions and work with that instead.
+///
+/// ## Memory safety
+///
+/// The type is `#[repr(transparent)]` for internal purposes only!
+/// No consumer crate may rely on the represenation of the struct!
+///
+/// ## References
+///
+///
+/// ### Bitcoin Core References
+///
+/// * [CScript definition](https://github.com/bitcoin/bitcoin/blob/d492dc1cdaabdc52b0766bf4cba4bd73178325d0/src/script/script.h#L410)
+///
+#[derive(PartialOrd, Ord, PartialEq, Eq, Hash)]
+#[repr(transparent)]
+pub struct Script(pub (in crate::blockdata::script) [u8]);
```

Comment:
I think this shouldn't need to be `pub (...)` because `from_{mut_}bytes` and `as_{mut_}bytes` should be sufficient and used everywhere else.

Reply:
Will fix as a follow up if that's ok to keep the re-ack trivial, I have to force push pick up the clippy fix.